packages/server-wallet/src/wallet/store.ts

```diff
@@ -382,11 +389,35 @@ export class Store {
         return this.ensureOpenChannelObjective(objective, tx);
       case 'CloseChannel':
         return this.ensureCloseChannelObjective(objective, tx);
+
+      case 'SubmitChallenge':
+        return this.ensureSubmitChallengeObjective(objective, tx);
       default:
         throw new StoreError(StoreError.reasons.unimplementedObjective);
     }
   }
 
+  private async ensureSubmitChallengeObjective(
+    objective: SubmitChallenge,
+    tx: Transaction
+  ): Promise<DBObjective> {
```

Comment:
I think that's a good point. This method is mostly based on the existing `ensure___Objective` methods that I copied it from 😛 

> If I understand correctly, it is going to (at least) swallow all errors that are not UniqueViolationErrors when inserting the objective. Why is that?

It should actually be doing the opposite (I hope)!  We want to ignore a `UniqueViolationErrors` since we have a `ON CONFLICT` clause in the db that handles it.```if (e.name !== 'UniqueViolationError') throw e;```

Reply:
Right, I must have missed the `!` first time around. This makes sense now.